modules/php/States/PreferencesTrait.php

```diff
@@ -0,0 +1,20 @@
+<?php
+namespace BANG\States;
+use BANG\Managers\Players;
+use BANG\Core\Notifications;
+
+trait PreferencesTrait
+{
+  /*
+   * changePreference: changes some preferences for specific player
+   */
+  public function changePreference($playerId, $pref, $value)
+  {
+    switch ($pref) {
+      case OPTION_GENERAL_STORE_LAST_CARD:
+        Players::setGeneralStorePref($playerId, $value);
```

Comment:
I would do $player = Players::getCurrent(); and $player->setGeneralStorePref($value) instead, following previous comments.